source/common/regular-expressions.js

```diff
@@ -57,25 +353,71 @@ module.exports = {
-    let flag = (multiline) ? 'm' : ''
-    let idRegExpString = global.config.get('zkn.idRE')
-    // Make sure the ID definitely has at least one
-    // capturing group to not produce errors.
-    if (!(/\(.+?\)/.test(idRegExpString))) {
-      idRegExpString = `(${idRegExpString})`
-    }
+  'getTableRE': function () {
+    return RegExp(
+      /^\|.+\|$/.source,
+      'i')
+  },
 
+  /**
+   * Returns a regular expression that matches inline frames.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getTableHeadingRE': function () {
     return RegExp(
-      idRegExpString,
-      'g' + flag
+      /(^[- ]+$)|(^[- +:]+$)|(^[- |:+]+$)/.source
     )
+  },
+
+  /**
+   * Returns a regular expression that matches tasks.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getTaskRE': function () {
+    return RegExp(
+      /^(\s*)([-+*]) \[( |x)\] /.source,
+      'g')
+  },
+
+  /**
+   * Returns a regular expression that matches URLs.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getUrlRE': function () {
+    return RegExp(
+      /^\[([^\]]+)\]\((.+?)\)|(((?:(?:aaas?|about|acap|adiumxtra|af[ps]|aim|apt|attachment|aw|beshare|bitcoin|bolo|callto|cap|chrome(?:-extension)?|cid|coap|com-eventbrite-attendee|content|crid|cvs|data|dav|dict|dlna-(?:playcontainer|playsingle)|dns|doi|dtn|dvb|ed2k|facetime|feed|file|finger|fish|ftp|geo|gg|git|gizmoproject|go|gopher|gtalk|h323|hcp|https?|iax|icap|icon|im|imap|info|ipn|ipp|irc[6s]?|iris(?:\.beep|\.lwz|\.xpc|\.xpcs)?|itms|jar|javascript|jms|keyparc|lastfm|ldaps?|magnet|mailto|maps|market|message|mid|mms|ms-help|msnim|msrps?|mtqp|mumble|mupdate|mvn|news|nfs|nih?|nntp|notes|oid|opaquelocktoken|palm|paparazzi|platform|pop|pres|proxy|psyc|query|res(?:ource)?|rmi|rsync|rtmp|rtsp|secondlife|service|session|sftp|sgn|shttp|sieve|sips?|skype|sm[bs]|snmp|soap\.beeps?|soldat|spotify|ssh|steam|svn|tag|teamspeak|tel(?:net)?|tftp|things|thismessage|tip|tn3270|tv|udp|unreal|urn|ut2004|vemmi|ventrilo|view-source|webcal|wss?|wtai|wyciwyg|xcon(?:-userid)?|xfire|xmlrpc\.beeps?|xmpp|xri|ymsgr|z39\.50[rs]?):(?:\/{1,3}|[a-z0-9%])|www\d{0,3}[.]|[a-z0-9.-]+[.][a-z]{2,4}\/)(?:[^\s()<>]|\([^\s()<>]*\))+(?:\([^\s()<>]*\)|[^\s`*!()[\]{};:'".,<>?«»“”‘’])))|([a-z0-9.\-_+]+?@[a-z0-9.\-_+]+\.[a-z]{2,7})$/.source,
+      'i')
+  },
+
+  /**
+   * Returns a regular expression that matches inline frames.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getWysiwygRE': function () {
+    return RegExp(
+      /__(.+?)__|_(.+?)_|\*\*(.+?)\*\*|\*(.+?)\*|^(#{1,6}) (.+?)$|^(?:\s*)> (.+)$|`(.+?)`/.source,
+      'gi')
+  },
+
+  /**
+   * Returns a regular expression that matches Zettelkasten IDs.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getZknTagRE': function () {
+    return RegExp(
+      /##?[^\s,.:;…!?"'`»«“”‘’—–@$%&*#^+~÷\\/|<=>[\](){}]+#?/.source,
```

Comment:
I remember that I've added a negative lookbehind to make sure the hashtag is either on the beginning of a line or preceded by a space, but then I don't know where this regexp is being used …

In general: It's a very good idea to collect all REs here, because there's so many potential issues you can spot with this!

Reply:
`.../markdown-editor/modes/spellchecker.js` and `.../markdown-editor/modes/markdown-zkn.js` used to have independent implementations of this regex.

Reply:
That's bad – did you merge them? (Because, after all, both are modes, so both must work on exactly the same basis)